sdk/scheduler/src/main/java/com/mesosphere/sdk/framework/SchedulerDriverFactory.java

```diff
@@ -94,53 +99,74 @@ public SchedulerDriver create(
         return createInternal(scheduler, frameworkInfo, masterUrl, credential, schedulerConfig.getMesosApiVersion());
     }
 
+    @VisibleForTesting
+    protected Function<MesosToSchedulerDriverAdapter, Mesos> startInternalCustom(
```

Comment:
Having this return a function instead of just running directly is overkill, as nothing's really using this asynchronously. How about just accepting the `MesosToSchedulerDriverAdapter` as a parameter and returning the `Mesos` object directly?